src/Blazor.FlexGrid/Components/Renderers/GridCellRenderer.cs

```diff
@@ -16,10 +22,30 @@ public GridCellRenderer(EditInputRendererTree editInputRendererTree)
         public override bool CanRender(GridRendererContext rendererContext)
             => true;
 
+
+
         protected override void BuildRendererTreeInternal(GridRendererContext rendererContext, PermissionContext permissionContext)
         {
             rendererContext.OpenElement(HtmlTagNames.TableColumn, rendererContext.CssClasses.TableCell);
 
+            Action onRowClickAction = null;
+            if (rendererContext.TableDataSet.GetType().GetGenericTypeDefinition() == typeof(TableDataSet<>))
+            {
+
+                var onRowClicked = OnRowClicked(rendererContext);
+
+                onRowClickAction = onRowClicked?.Invoke(rendererContext);
+                
+                if (!rendererContext.IsActualItemEdited)
+                    rendererContext.AddOnClickEvent(
```

Comment:
Indeed this is much better... but with this extra improvement: putting the declaration and initialization of "localActualItem" outside of the lambda expression. Otherwise, when clicking a row, localActualItem doesn't point to the corresponding item but to the latest value assigned to rendererContext.ActualItem, that is the item of the last row in the table. In other words, clicking any row always executes the action corresponding to the last row.

Reply:
var localActualItem = rendererContext.ActualItem; This is the magic line and it will be working with correct item. I use this approach in other scenarios in FlexGrid

Reply:
Yes this `var localActualItem = rendererContext.ActualItem;` line must be in `GetEventHandlerValue` method